### ISSUE TITLE
Load Past Emails Features

### ISSUE THREAD

**Issue by bhorowitz** [learning-unlimited/ESP-Website#2006]
Since programs often re-use email templates from past programs, it might make sense to have a "import past email" feature similar to that for teacher's importing past classes. Currently it is very difficult to find the html for past emails, and usually involves remembering the exact title of it or sifting through all emails.

**Comment by benjaminjkraft** [comment]
We do even save all the past emails, so this shouldn't be hard to do.

**Comment by mgersh** [comment]
We already have a feature somewhat like this in the "Previous e-mails" tab on the comm panel. (I only know about this because its data load has been erroring on Yale's site for months and I found it in serverlog.) Sounds like @bhorowitz possibly knows about this feature and wants something better, though?

**Comment by bhorowitz** [comment]
@mgersh : Right, that feature does exist, but you need to remember the exact email name (or at least what it begins with). I was thinking of some sort of structure where you can select the program and see all the emails that were sent out for that program.

<img width="712" alt="screen shot 2016-03-13 at 1 23 34 pm" src="https://cloud.githubusercontent.com/assets/6059772/13731122/fac462f8-e91e-11e5-91c0-e03b3a8199fb.png">

**Comment by willgearty** [comment]
Also, the current implementation doesn't permit you to change the recipients of the email.

**Comment by Supriyo760** [comment]
hi @willgearty 
It sounds like there are two distinct areas for improvement here:

1. **Better Discovery:** The current "Previous e-mails" autocomplete is severely limited if you don't remember the exact subject line. We could replace or supplement this with a structured UI (perhaps a modal or a nested dropdown) that groups past emails first by **Program**, and then lists the emails within that program. 
2. **Better Reusability:** As Will mentioned, the current flow doesn't permit changing recipients when an email is "loaded" this way. We'd need to ensure the import action populates the compose form's subject/body/settings fields but leaves the recipient selection open for modification.

Would the ideal approach be an "Import from previous email" button that opens a modal where you can select `[Program] -> [Email]`, which then injects that email's content (but *not* its old recipients) into your current draft? 

If this approach sounds like what you're all envisioning, I'd love to work on this.

**Comment by willgearty** [comment]
I think that sounds good @Supriyo760
